flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/DefaultCheckpointStatsTracker.java

```diff
@@ -256,7 +256,13 @@ private void logCheckpointStatistics(AbstractCheckpointStats checkpointStats) {
                             .setAttribute("checkpointId", checkpointStats.getCheckpointId())
                             .setAttribute("fullSize", checkpointStats.getStateSize())
                             .setAttribute("checkpointedSize", checkpointStats.getCheckpointedSize())
-                            .setAttribute("checkpointStatus", checkpointStats.getStatus().name()));
+                            .setAttribute("checkpointStatus", checkpointStats.getStatus().name())
+                            .setAttribute(
+                                    "isUnaligned",
+                                    Boolean.toString(checkpointStats.isUnalignedCheckpoint()))
+                            .setAttribute(
+                                    "checkpointType",
+                                    checkpointStats.getProperties().getCheckpointType().getName()));
```

Comment:
in this case I assume that the name will always be "Checkpoint". I assume savepoints do not come through here. If save points can come through here then the span.builder first line should probable be "Snapshot" rather than "Checkpoint".
I am thinking that the SnapshotType's SharingFilesStrategy - would be good to include here.

Reply:
The potential values here are:
```
    public static final CheckpointType CHECKPOINT =
            new CheckpointType("Checkpoint", SharingFilesStrategy.FORWARD_BACKWARD);

    public static final CheckpointType FULL_CHECKPOINT =
            new CheckpointType("Full Checkpoint", SharingFilesStrategy.FORWARD);
            
    public static SavepointType savepoint(SavepointFormatType formatType) {
        return new SavepointType("Savepoint", PostCheckpointAction.NONE, formatType);
    }

    public static SavepointType terminate(SavepointFormatType formatType) {
        return new SavepointType("Terminate Savepoint", PostCheckpointAction.TERMINATE, formatType);
    }

    public static SavepointType suspend(SavepointFormatType formatType) {
        return new SavepointType("Suspend Savepoint", PostCheckpointAction.SUSPEND, formatType);
    }
```

Savepoints do end up here, and unfortunately the `Checkpoint` name here is a bit overloaded. However this is also consistent with other places in the code, especially metrics/logs. There savepoints are also reported/logged as checkpoints. So I'm not sure that I would be fine the to change the trace's `Checkpoint` name  here to `Snapshot`.

I think `SharingFilesStrategy` is superfluous. Each snapshot type is named uniquely, so you can deduce sharing strategy based on the name/type.